File: AndroidPromotionUI/src/main/java/com/ttcn/promotionsdk/ui/theme/PromotionSDKTheme.kt
package com.ttcn.promotionsdk.ui.theme

import com.ttcn.promotionsdk.ui.theme.token.ButtonToken
import com.ttcn.promotionsdk.ui.theme.token.DiscountBadgeToken
import com.ttcn.promotionsdk.ui.theme.token.ListItemToken
import com.ttcn.promotionsdk.ui.theme.token.SearchBarToken
import com.ttcn.promotionsdk.ui.theme.token.TabChipToken
import com.ttcn.promotionsdk.ui.theme.token.TabUnderlineToken

/**
 * Cấu hình theme của SDK. Đối ứng 1-1 với `PromotionSDKTheme.swift` bên iOS: cùng sáu token, cùng
 * tên field, cùng ngữ nghĩa `null` = **giữ mặc định của SDK** cho nhóm đó.
 *
 * Trước đây Android còn có `PromotionThemeConfig` — bản sao đúng sáu field này, kèm
 * `PromotionSDKTheme.from(config)` và `toThemeConfig()` chuyển qua lại mà không type nào mang thêm
 * thông tin gì. Đã gộp về một, khớp với iOS.
 */
data class PromotionSDKTheme(
    val buttonToken: ButtonToken? = null,
    val searchBarToken: SearchBarToken? = null,
    val listItemToken: ListItemToken? = null,
    val tabChipToken: TabChipToken? = null,
    val tabUnderlineToken: TabUnderlineToken? = null,
    val discountBadgeToken: DiscountBadgeToken? = null,
)

File: iosPromotionSDK/PromotionSDKUI/Theme/PromotionSDKTheme.swift
//
//  PromotionSDKTheme.swift
//  PromotionSDK
//
//  Cấu hình theme công khai. Chỉ UIColor/CGFloat để không kéo PRMDesignKit vào module interface.
//  Đối ứng `PromotionSDKTheme.kt`. Sáu token ở `Token/`, serialize ở `PromotionThemeJson.swift`,
//  codec màu ở `ThemeHex.swift` — chia thư mục giống Android.
//

import UIKit

/// Cấu hình theme tổng cho SDK. Nhóm nil = giữ toàn bộ default của SDK cho nhóm đó.
public struct PromotionSDKTheme {
    public var buttonToken: ButtonToken?
    public var searchBarToken: SearchBarToken?
    public var listItemToken: ListItemToken?
    public var tabChipToken: TabChipToken?
    public var tabUnderlineToken: TabUnderlineToken?
    public var discountBadgeToken: DiscountBadgeToken?

    public init(buttonToken: ButtonToken? = nil,
                searchBarToken: SearchBarToken? = nil,
                listItemToken: ListItemToken? = nil,
                tabChipToken: TabChipToken? = nil,
                tabUnderlineToken: TabUnderlineToken? = nil,
                discountBadgeToken: DiscountBadgeToken? = nil) {
        self.buttonToken = buttonToken
        self.searchBarToken = searchBarToken
        self.listItemToken = listItemToken
        self.tabChipToken = tabChipToken
        self.tabUnderlineToken = tabUnderlineToken
        self.discountBadgeToken = discountBadgeToken
    }
}
